scripts/tag_reconciler.py

```diff
@@ -0,0 +1,144 @@
+        if not os.path.exists(VERSIONS_FILE):
+            logger.error(f"{VERSIONS_FILE} does not exist")
+            sys.exit(1)
+
+        try:
+            with open(VERSIONS_FILE, "r") as f:
+                data = yaml.load(f)
+            return data or {}
+        except Exception as e:
+            logger.error(f"Error reading {VERSIONS_FILE}: {e}")
+            return {}
+
+    def tag_exists(self, repo: str, tag: str) -> bool:
+        try:
+            github_repo = self.github_client.get_repo(repo)
+            github_repo.get_git_ref(f"tags/{tag}")
+            return True
+        except Exception as e:
+            logger.error(f"Error checking tag {tag} for {repo}: {e}")
+            return False
+
+    def create_tag(self, repo: str, commit_sha: str, tag: str) -> bool:
+        message = f"Version {tag} - Tagged by CAPBCOA CI"
+        
+        try:
+            github_repo = self.github_client.get_repo(repo)
+            tag_obj = github_repo.create_git_tag(
+                tag=tag,
+                message=message,
+                object=commit_sha,
+                type="commit"
+            )
+            
+            github_repo.create_git_ref(f"refs/tags/{tag}", tag_obj.sha)
+            logger.info(f"Created tag {tag} for {repo}")
+            return True
+        except Exception as e:
+            logger.error(f"Error creating tag {tag} for {repo}: {e}")
+            return False
+
+    def extract_base_repo(self, full_repo_name: str) -> str:
+        parts = full_repo_name.split("/")
+        if len(parts) > 2:
+            return f"{parts[0]}/{parts[1]}"
+        return full_repo_name
+
+    def reconcile_tag(self, version: dict[str, Any]) -> bool:
+        version_name = version.get("name", "")
+        if not version_name:
+            logger.error("Version entry missing 'name' field")
+            return False
+
+        logger.info(f"Processing version {version_name}")
+        success = True
+
+        repositories = version.get("repositories", {})
+        if not repositories:
+            logger.warning(f"No repositories found for version {version_name}")
+            return True
+
+        for repo_name, repo_info in repositories.items():
+            commit_sha = repo_info.get("commit_sha", "")
+
+            if not re.match(r"^openshift/", repo_name) or not commit_sha:
+                logger.warning(f"Skipping repository {repo_name}")
+                continue
+
+            base_repo = self.extract_base_repo(repo_name)
+            if self.tag_exists(base_repo, version_name):
+                logger.info(f"Tag {version_name} already exists for {base_repo}")
+                continue
+            if not self.create_tag(base_repo, commit_sha, version_name):
+                logger.error(f"Failed to create tag {version_name} for {base_repo}")
+                success = False
+
+        return success
+
+    def reconcile_tags(self) -> bool:
+        versions_data = self.read_versions_file()
```

Comment:
Can we document somwehere the expected format of this file? With a test would be ideal, if complicated a README would do

Reply:
Actually I think defining a type for each version and the version file could be clearer.

As generic remark I think we need in the version (and candidate version) file the git repo/git ref pairs. This could come both as a tag or as a commit ref, so it'd be the same for owned and not owned repos.

Then we could use another utility to transform this values in images if need be (to fetch the right sha)